### PR TITLE
Tanner/ganalytics fix vnext

### PR DESCRIPTION
**Pull request by tanneryould**
# Description

Removes white spaces from analytics event names and fixes a bug where the telemetry box would sometimes display during automated testing.

## Type of change

- [x] Bug fix

**Platforms tested on**:

- [x] macOS

## Checklist

- [x] Runs and compiles on all active platforms
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code

### CODE REVIEW

File: ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp [hunk 1]
```diff
@@ -50,7 +50,9 @@ GAnalytics::~GAnalytics()
 void GAnalytics::init()
 {
   generateClientId();
-  if (m_apiSecret.isEmpty() || m_measurementId.isEmpty())
+
+  // If this is an Esri daily build or a local user build, disable analytics
+  if (m_apiSecret.isEmpty() || m_measurementId.isEmpty() || m_apiSecret == "GANALYTICS_API_KEY" || m_measurementId == "GANALYTICS_STREAM_ID")
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp:55]: Why would `m_apiSecret == "GANALYTICS_API_KEY"` ever happen? We should get to the bottom of that. This seems like a workaround.

File: ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp [hunk 2]
```diff
@@ -78,7 +81,12 @@ void GAnalytics::init()
 #endif // CPP_VIEWER / QML_VIEWER
 
   const auto os = QOperatingSystemVersion::current();
-  m_defaultParameters.insert("operating_system", QString{"%1 %2.%3"}.arg(os.name(), os.majorVersion(), os.minorVersion()));
+  m_defaultParameters.insert("operating_system", QString{"%1 %2.%3"}.arg(os.name()).arg(os.majorVersion()).arg(os.minorVersion()));
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp:84]: This is an [anti-pattern](https://github.com/KDE/clazy/blob/master/docs/checks/README-qstring-arg.md). 

Something that jumps out to me is `os.majorVersion()` and `os.minorVersion()`. Are those ints? In that case, you may want this instead:
```suggestion
  m_defaultParameters.insert("operating_system", QString{"%1 %2.%3"}.arg(os.name(), QString::number(os.majorVersion()), QString::number(os.minorVersion())));
```

(warning - syntax not validated - be sure to test it)

File: ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp [hunk 3]
```diff
@@ -78,7 +81,12 @@ void GAnalytics::init()
...
+
+  m_googleAnalyticsUrl = QString{"https://google-analytics.com/mp/collect?api_secret=%1&measurement_id=%2"}.arg(m_apiSecret).arg(m_measurementId);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp:86]: Same comment, use multi-arg `arg` and see if either of these args are numerics. Use QString::number in that case.

File: ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp (hunk 1 above)
Comment (tanneryould) [reply ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp:55]: If `GANALYTICS_API_KEY` isn't defined then `QUOTE(GANALYTICS_API_KEY)` evaluates to `"GANALYTICS_API_KEY"`. If it's defined but `GANALYTICS_API_KEY` wasn't set on the command line, it evaluates to `""`. What I should do instead is move the defines outside of the daily build block and make sure it's defined regardless.

**Review by JamesMBallard — APPROVED** [review]

**Pull request merged**